File: LeetCode/Remove_Dulicates_from_Sorted_Array_26.py
#LeetCode link: https://leetcode.com/problems/remove-duplicates-from-sorted-array/

class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        #new list with deleted dublicates
        new_nums = sorted(list(set(nums)))
        
        new_length = len(new_nums)
        old_length = len(nums)
        
        #copying new array to nums
        for index, num in enumerate(new_nums):
            nums[index] = num
        
        #deteleting old elements
        for _ in range(new_length, old_length):
            del nums[new_length]
        
        return new_length

File: LeetCode/Remove_Dulicates_from_Sorted_Array_26.c
#LeetCode link: https://leetcode.com/problems/remove-duplicates-from-sorted-array/

int removeDuplicates(int* nums, int nums_size) {
    int nums_count = 0; //amount of unique elements
    for(int i = 0; i < nums_size; ++i){
        nums[nums_count] = nums[i];
        while(nums[nums_count] == nums[i]) 
            ++i;
        nums_count += 1;
        i -= 1;
    }
    return nums_count;
}